src/indev/lv_indev.c

```diff
@@ -1267,6 +1246,23 @@ static void indev_proc_press(lv_indev_t * indev)
  */
 static void indev_proc_release(lv_indev_t * indev)
 {
+    if((indev->pointer.last_point.x != indev->pointer.act_point.x ||
+        indev->pointer.last_point.y != indev->pointer.act_point.y)) {
+        lv_obj_t ** layer = &indev->disp->sys_layer;
+        lv_obj_t ** last = &indev->pointer.last_hovered;
+        lv_obj_t * hved = NULL;
+        uint8_t layers = 4;
+        while(!hved && layers) {
+            hved = lv_indev_search_obj(*layer++, &indev->pointer.act_point);
+            if(*last != hved) {
+                if(lv_obj_is_valid(hved)) lv_obj_send_event(hved, LV_EVENT_HOVER_OVER, indev_act);
+                if(lv_obj_is_valid(*last)) lv_obj_send_event(*last, LV_EVENT_HOVER_LEAVE, indev_act);
```

Comment:
`lv_obj_is_valid` shouldn't be used here becasue it's slow and sometimes gives incorrect result. Instead `indev->pointer.last_hovered` shall be `NULL`-ed explicitly [here](https://github.com/lvgl/lvgl/blob/master/src/indev/lv_indev.c#L1567-L1569) and [here](https://github.com/lvgl/lvgl/blob/master/src/indev/lv_indev.c#L1388).

Reply:
Settled.